lang/cs/Org.Apache.REEF.Tests/Functional/Failure/User/SendTaskMessageExceptionTest.cs

```diff
@@ -0,0 +1,165 @@
+            IObserver<IFailedEvaluator>
+        {
+            private readonly IEvaluatorRequestor _requestor;
+
+            [Inject]
+            private TestSendTaskMessageExceptionDriver(IEvaluatorRequestor requestor)
+            {
+                _requestor = requestor;
+            }
+
+            public void OnNext(IDriverStarted value)
+            {
+                _requestor.Submit(_requestor.NewBuilder().Build());
+            }
+
+            public void OnNext(IAllocatedEvaluator value)
+            {
+                value.SubmitTask(
+                    TaskConfiguration.ConfigurationModule
+                        .Set(TaskConfiguration.Identifier, TaskId)
+                        .Set(TaskConfiguration.Task, GenericType<TestTask>.Class)
+                        .Set(TaskConfiguration.OnSendMessage, GenericType<SendTaskMessageExceptionHandler>.Class)
+                        .Build());
+            }
+
+            /// <summary>
+            /// Throwing an Exception in a task message handler will result in a Failed Evaluator.
+            /// </summary>
+            public void OnNext(IFailedEvaluator value)
+            {
+                Assert.Equal(1, value.FailedContexts.Count);
+                Assert.NotNull(value.EvaluatorException.InnerException);
+                Assert.True(value.EvaluatorException.InnerException is TestSerializableException,
+                    "Unexpected type of evaluator exception: " + value.EvaluatorException.InnerException.GetType());
+                Assert.Equal(ExpectedExceptionMessage, value.EvaluatorException.InnerException.Message);
+
+                Logger.Log(Level.Info, ReceivedFailedEvaluator);
+            }
+
+            public void OnNext(IFailedContext value)
+            {
+                throw new Exception("The Driver does not expect a Failed Context message.");
+            }
+
+            public void OnError(Exception error)
+            {
+                throw new NotImplementedException();
+            }
+
+            public void OnCompleted()
+            {
+                throw new NotImplementedException();
+            }
+        }
+
+        /// <summary>
+        /// A Context message source that throws an Exception.
+        /// </summary>
+        private sealed class SendTaskMessageExceptionHandler : ITaskMessageSource
+        {
+            private int counter;
+
+            [Inject]
+            private SendTaskMessageExceptionHandler()
+            {
+            }
+
+            public Optional<TaskMessage> Message
+            {
+                get
+                {
+                    counter++;
+                    if (counter == 2)
+                    {
+                        throw new TestSerializableException(ExpectedExceptionMessage);
+                    }
+                    return Optional<TaskMessage>.Empty();
+                }
+            }
+        }
+
+        private sealed class TestTask : WaitingTask
```

Comment:
We expect evaluator to fail on 2nd attempt to send heartbeat (due to exception thrown in `OnSendMessage`). There's no need to signal the task